### PR TITLE
[SPARK-25478][SQL][TEST] Refactor CompressionSchemeBenchmark to use main method

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Refactor `CompressionSchemeBenchmark` to use main method.
Generate benchmark result:
```sh
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.columnar.compression.CompressionSchemeBenchmark"
```

## How was this patch tested?

manual tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/compression/CompressionSchemeBenchmark.scala [hunk 1]
```diff
@@ -318,28 +229,17 @@ object CompressionSchemeBenchmark extends AllCompressionSchemes {
...
   }
 
-  def main(args: Array[String]): Unit = {
-    bitEncodingBenchmark(1024)
-    shortEncodingBenchmark(1024)
-    intEncodingBenchmark(1024)
-    longEncodingBenchmark(1024)
-    stringEncodingBenchmark(1024)
+  override def benchmark(): Unit = {
+    runBenchmark("encoding benchmark") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/compression/CompressionSchemeBenchmark.scala]: How about `Compression Scheme Benchmark`?

**Comment by dongjoon-hyun** [comment]
Could you add `[SQL]` at the title?

**Comment by dongjoon-hyun** [comment]
Hi, @wangyum . If you don't mind, could you review and merge [my PR](https://github.com/wangyum/spark/pull/10) to your branch? The BM title is updated and latest OpenJDK on AWS is used.

**Comment by wangyum** [comment]
@dongjoon-hyun Thanks a lot.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

Thank you, @wangyum .

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**